### DISCUSSION TITLE
Get current route name

### DISCUSSION THREAD

**Discussion by notramo** [martenframework/marten#163] [Ideas]
It can be useful for highlighting currently active route in navigation sections.

When using route groups (e.g. `posts:list`, `posts:detail`), it should match `posts` for both `posts:list` and `posts:detail`, so if the route is inside the group, it matches the route group name.

If `Marten::Routing::Map#resolve` would return the route name too, that could be used.

Also, a template tag would be useful: `{% current_route_matches "posts" %}`.

**Comment by ellmetha** [comment]
Hey! 👋 

From my experience, this is something that varies greatly from one project to another. Inferring the current navigation item from route namespaces could work for some projects, but for some others with different behaviours and implementations, it could not work at all afaics!

This is also something that can be implemented very easily by defining a concern module and having handlers define which "navigation section" they are associated with. This is the approach used by the [Marten Realworld](https://github.com/martenframework/realworld/blob/main/src/handlers/concerns/nav_bar_activeable.cr) demo application. For example:

```crystal
module NavBarActiveable
  macro included
    class_getter nav_bar_item : String?

    extend NavBarActiveable::ClassMethods

    before_render :add_nav_bar_item_to_context
  end

  module ClassMethods
    def nav_bar_item(item : String | Symbol)
      @@nav_bar_item = item.to_s
    end
  end

  private def add_nav_bar_item_to_context
    context[:nav_bar_item] = self.class.nav_bar_item
  end
end

class MyHandler < Marten::Handler
  include NavBarActiveable

  nav_bar_item :posts
end
```

Then in your templates you can simply make use of the `nav_bar_item` variable to determine which navigation item is the active one.

But we could indeed ensure the resolved route name is added to `Marten::Routing::Match` objects so that projects can make use of this information if necessary.

**Reply by notramo** [reply]
I was thinking about this approach too, but have not found enough info in the documentation. I use CRUD classes (e.g. `Marten::Handlers::RecordList`, and I thought it would be possible to add the active item if I used simply `Marten::Handler` and manually added the item.
I recommend documenting this 2 use cases in the "How to" section:
- How to add custom data to template context when using CRUD classes?
- Specific example of navigation link highlighting.

**Reply by ellmetha** [reply]
> I was thinking about this approach too, but have not found enough info in the documentation.

Thanks a lot for this feedback! `before_render` callbacks are [documented](https://martenframework.com/docs/handlers-and-http/callbacks#before_render) but your message seems to indicate that we are probably missing some guidelines somewhere else on how to easily customize template variables and make those easily findable. Maybe we could add a dedicated how-to 👍 

> Specific example of navigation link highlighting.

I guess we could showcase the navigation example in this how-to I was mentioning.
